Bidsketch/docsketch
# Install test frameworks

This  PR adds a test framework comprised of the following:

1) [Mocha test runner](https://mochajs.org/)
2) [Chai assertions](http://chaijs.com/api/assert/)
3) [Enzyme React testing utilities](http://airbnb.io/enzyme/)
4) [React Test Addon Utils](https://facebook.github.io/react/docs/test-utils.html)
5) [Sinon test spies, stubs, and mocks](http://sinonjs.org/)
6) [Jsdom javascript DOM-like API without needing a browser](https://github.com/tmpvar/jsdom)

To test:

1) run npm test
2) See that two tests are passing
3) View test/javascripts/hello_world_test.es6.js to see example test that was run.

[Trello Card Requirements: Setup JS Testing](https://trello.com/c/vaBVRY7H)

1) A developer should be able to write tests using the assert syntax

_This is supported through Chai._

3) A developer should be able to test the output of a React component

_This is supported through Enzyme._

5) A developer should be able to test how an event changes a React component

_This is supported through Enzyme._

7) A developer should be able to test a javascript function

_This is supported through Mocha._

9) A developer should be able to run the entire test suite quickly

_This is supported through npm script "test" i.e. `npm test`._

11) A developer should be able to run a single test file quickly

_[This is supported through mocha](http://jaketrent.com/post/run-single-mocha-test/)_

13) A developer should be able to run a single test quickly

 _[This is supported through mocha](http://jaketrent.com/post/run-single-mocha-test//)_

14) A developer should be able to allow the test suite to run on Codeship

_This is supported through npm script "test" i.e. `npm test`._

15) A developer should be able to use fixture data in their tests

 _* This is supported by commonJS, you can create dummy JS classes as fixtures._
